### ISSUE TITLE
Feature Request: cloud shadows

### ISSUE THREAD

**Issue by MJacred** [Lexpartizan/Godot_sky_shader#6]
Maybe implementing them like the light cookies as used in Unity
* https://docs.unity3d.com/Manual/Cookies.html
* discussion on Reddit for Godot: https://www.reddit.com/r/godot/comments/8tbsdo/light_cookies_in_godot/
* in Godot repo, they mentioned Stencils and Decals
  * https://github.com/godotengine/godot/issues/2002
  * https://github.com/godotengine/godot/issues/19101
  * https://github.com/godotengine/godot/issues/21185
  * https://github.com/godotengine/godot/issues/23721

Only decal implementation found:  
Screen-Space-Decals: https://github.com/Mr-Slurpy/Screen-Space-Decals

In-Engine support for decals might come with Godot 4...

The shadows probably won't have to be 100% accurate, because that's difficult to tell, but they should match with the cloud coverage

**Comment by Lexpartizan** [comment]
Cloud coverage is now stored in a texture, so there is a base for work.
But this is working with light, so I have no idea how to project a shadow.
In real life, we don't see shadows from specific clouds.
It could be a little easier to reduce the light when the center of the Sun is behind the clouds, but I don't know how to do it yet either, becouse i change light with script, no shader.

**Comment by MJacred** [comment]
Hm... I believe you can make out the shadow in real life, if the sun is at its zenith or close to it... ie. the sun is shining really strong onto the Earth surface. I'll do some research in the following days

**Comment by Lexpartizan** [comment]
Yes, this happens, but very rarely, in General, the overall illumination changes.
